projects/storefrontlib/src/lib/product/facade/product.service.ts

```diff
@@ -0,0 +1,39 @@
+import { Injectable } from '@angular/core';
+import { Store, select } from '@ngrx/store';
+import { Observable } from 'rxjs';
+
+import * as fromStore from '../store';
+import { map, tap, filter, take } from 'rxjs/operators';
+
+@Injectable()
+export class ProductService {
+  constructor(private store: Store<fromStore.ProductsState>) {}
+
+  get(productCode: string): Observable<any> {
+    return this.store.pipe(
+      select(fromStore.getSelectedProductFactory(productCode))
+    );
+  }
+
+  isProductLoaded(requestedProductCode: string): Observable<boolean> {
```

Comment:
isProductLoaded(productCode) {
    return this.store.pipe(
      select(fromStore.getSelectedProductFactory(requestedProductCode)),
      map(product => {
        if (product) {
          return Object.keys(product).length !== 0;
        } else {
          return false;
        }
      })
}

Then, we need add one more function: loadProductData.
ProductCarousel calls loadProductData.